droidlet/tools/hitl/dashboard_app/backend/dashboard_aws_helper.py

```diff
@@ -92,8 +97,62 @@ def get_run_info_by_id(batch_id: int):
     """
     local_fname = _download_file(f"job_management_records/{batch_id}.json")
     if local_fname is None:
-        return f"cannot find run info with id {batch_id}"
+        return f"cannot find run info with id {batch_id}", 404
     f = open(local_fname)
     json_data = json.load(f)
     f.close()
-    return json_data
+    return json_data, None
+
+
+def get_interaction_sessions_by_id(batch_id: int):
+    session_list = []
+    s3_bucket = s3.Bucket(S3_BUCKET_NAME)
+    prefix = f"{batch_id}/interaction/"
+
+    for obj in s3_bucket.objects.filter(Prefix=prefix):
+        session_name = obj.key
+        left_idx = session_name.index(prefix) + len(prefix)
+        right_idx = session_name.index("/logs.tar.gz")
+        session_list.append(session_name[left_idx:right_idx])
+    return session_list
+
+
+def get_dataset_indices_by_id(batch_id: int):
+    """
+    For a run specied by the batch id,
+    get the newly added dataset data points' start index and end index from meta.txt
+    """
+    local_fname = _download_file(f"{batch_id}/meta.txt")
+    if local_fname is None:
+        return f"cannot find meta.txt with id {batch_id}", 404
+    meta_txt_content = _read_file(local_fname)
+    meta_txt_splitted = meta_txt_content.split("\n")
+    return [int(meta_txt_splitted[0]), int(meta_txt_splitted[-1])], None
+
+
+def get_dataset_version_list_by_pipeline(pipeline: str):
```

Comment:
I added this file to S3: `https://s3.console.aws.amazon.com/s3/object/droidlet-hitl?region=us-west-2&prefix=nsp_retrain_meta.json`, which contains all meta info of NSP retrain pipeline. You can retrieve info like dataset version list there. Please take a look.

Reply:
Thanks for adding the meta data file!

Reply:
Hi @snyxan I updated the aws helper to use meta data json file directly rather than crawling the aws bucket. One question that I want to ask your opinion about is do we actually need to distinguish the latest version - my idea is since the version is always ascending, we can just sort it to get the latest version.

Reply:
This is just pointer to the one we use in production. We can pick another name like stable/prod, etc.